lib/rspec/mocks/proxy.rb

```diff
@@ -376,22 +376,22 @@ def original_method_handle_for(message)
     protected
 
       def original_unbound_method_handle_from_ancestor_for(message)
-        method_double = @method_doubles.fetch(message) do
-          # The fact that there is no method double for this message indicates
-          # that it has not been redefined by rspec-mocks. We need to continue
-          # looking up the ancestor chain.
-          return superclass_proxy &&
-            superclass_proxy.original_unbound_method_handle_from_ancestor_for(message)
+        # The fact that there is no method double for this message indicates
+        # that it has not been redefined by rspec-mocks. We need to continue
+        # looking up the ancestor chain.
+        if @method_doubles.include?(message)
+          @method_doubles.fetch(message).original_method.unbind
+        else
+          superclass_proxy\
+            && superclass_proxy.original_unbound_method_handle_from_ancestor_for(message)
         end
-
-        method_double.original_method.unbind
       end
```

Comment:
Why the change to this method?  It looks to me like what you have is semantically equivalent but slightly less efficient.  After all, for the case where `if @method_doubles.include?(message); @method_doubles.fetch(message)` is going to be less efficient than just `@method_doubles.fetch(message)` like we had before.